packages/client/hooks/useAnimatedSpotlightSource.ts

```diff
@@ -0,0 +1,67 @@
+import useAtmosphere from '~/hooks/useAtmosphere'
+import clientTempId from '~/utils/relay/clientTempId'
+import {BezierCurve} from '~/types/constEnums'
+import {Times} from 'parabol-client/types/constEnums'
+import {Elevation} from '~/styles/elevation'
+import cloneReflection from '~/utils/retroGroup/cloneReflection'
+import {PortalStatus} from '~/hooks/usePortal'
+import {MutableRefObject, useLayoutEffect, useRef} from 'react'
+import StartDraggingReflectionMutation from '~/mutations/StartDraggingReflectionMutation'
+
+const useAnimatedSpotlightSource = (
+  portalStatus: PortalStatus,
+  reflectionId: string | undefined,
+  dragIdRef: MutableRefObject<string | undefined>
+) => {
+  const atmosphere = useAtmosphere()
+  const sourceRef = useRef<HTMLDivElement | null>(null)
+  const sourceCloneRef = useRef<HTMLDivElement | null>(null)
+
+  useLayoutEffect(() => {
+    const {current: source} = sourceRef
+    const {current: sourceClone} = sourceCloneRef
+    // wait for the modal to enter to get the source's bbox
+    if (portalStatus !== PortalStatus.Entered || !sourceClone || !reflectionId || !source) return
+    const sourceBbox = source.getBoundingClientRect()
+    const sourceCloneBbox = sourceClone.getBoundingClientRect()
+    const {style: sourceStyle} = source
+    sourceStyle.opacity = '0' // hide source while animating sourceClone into modal
+    const clone = cloneReflection(sourceClone, reflectionId)
+    const {style: cloneStyle} = clone
+    const {left: startLeft, top: startTop} = sourceCloneBbox
+    const {left: endLeft, top: endTop} = sourceBbox
+    const roundedEndTop = Math.round(endTop) // fractional top pixel throws off transform position
+    cloneStyle.left = `${startLeft}px`
+    cloneStyle.top = `${startTop}px`
```

Comment:
In the Loom, you mentioned that we don't want to set the initial left & top. However, `cloneReflection` creates an element without defining its left or top; without doing this here, the clone will be at the top left of the screen.

Reply:
i think i messed up.
setting an initial left, top is AOK, we just want to be sure we don't update the left, top